src/linear_algebra/matrix/diagonal.lean

```diff
@@ -60,8 +60,9 @@ variables {K : Type u} [field K] -- maybe try to relax the universe constraint
 lemma ker_diagonal_to_lin' [decidable_eq m] (w : m → K) :
   ker (diagonal w).to_lin' = (⨆i∈{i | w i = 0 }, range (linear_map.std_basis K (λi, K) i)) :=
 begin
-  rw [← comap_bot, ← infi_ker_proj],
-  simp only [comap_infi, (ker_comp _ _).symm, proj_diagonal, ker_smul'],
+  rw [← comap_bot, ← infi_ker_proj, comap_infi],
+  have := λ i : m, ker_comp (to_lin' (diagonal w)) (proj i),
+  simp only [comap_infi, ← this, proj_diagonal, ker_smul'],
```

Comment:
I'm curious what makes this no longer work.

Reply:
Some of these elaboration issues are still very mysterious to me as well!